source/detail/serialization/xlsx_producer.hpp

```diff
@@ -169,19 +170,34 @@ class xlsx_producer
 
     void write_namespace(const std::string &ns, const std::string &prefix);
 
-    template<typename T>
+    // std::string attribute name
+    // not integer or float type
+    template <typename T, typename = typename std::enable_if<!std::is_convertible<T, double>::value>::type>
     void write_attribute(const std::string &name, T value)
     {
         current_part_serializer_->attribute(name, value);
     }
 
-    template<typename T>
+    void write_attribute(const std::string &name, double value)
+    {
+        current_part_serializer_->attribute(name, converter_.serialise(value));
+    }
+
+    // qname attribute name
+    // not integer or float type
+    template <typename T, typename = typename std::enable_if<!std::is_convertible<T, double>::value>::type>
     void write_attribute(const xml::qname &name, T value)
     {
         current_part_serializer_->attribute(name, value);
     }
 
-    template<typename T>
+    void write_attribute(const xml::qname &name, double value)
+    {
+        current_part_serializer_->attribute(name, converter_.serialise(value));
+    }
```

Comment:
the little dance going on here is redirecting all numbers (types convertible to double) through the untemplated functions with double as the parameter. Without the enable if on the general case, the overload would only catch *exact* matches (i.e. double but not float/int).